pkg/storage/memory/memory_test.go

```diff
@@ -22,98 +23,231 @@ func TestMemdbStorage(t *testing.T) {
 	test.RunAllTests(t, ds)
 }
 
-func TestStaticTupleIteratorNoRace(t *testing.T) {
-	now := timestamppb.Now()
-
-	iter := &staticIterator{
-		records: []*storage.TupleRecord{
+func TestStaticTupleIterator(t *testing.T) {
+	t.Run("empty_iterator", func(t *testing.T) {
+		tests := []struct {
+			name  string
+			mixed bool
```

Comment:
Super minor nit:

```suggestion
			interleaveNextAndHeadCalls bool
```

or something like that